cpp/include/cudf_test/column_wrapper.hpp

```diff
@@ -183,12 +239,13 @@ rmm::device_buffer make_elements(InputIterator begin, InputIterator end)
 template <typename ValidityIterator>
 std::vector<bitmask_type> make_null_mask_vector(ValidityIterator begin, ValidityIterator end)
 {
-  cudf::size_type size = std::distance(begin, end);
-  auto num_words       = cudf::bitmask_allocation_size_bytes(size) / sizeof(bitmask_type);
-  std::vector<bitmask_type> null_mask(num_words, 0);
-  for (auto i = 0; i < size; ++i) {
-    if (*(begin + i)) { set_bit_unsafe(null_mask.data(), i); }
-  }
+  auto const size      = cudf::distance(begin, end);
+  auto const num_words = cudf::bitmask_allocation_size_bytes(size) / sizeof(bitmask_type);
+
+  auto null_mask = std::vector<bitmask_type>(num_words, 0);
```

Comment:
For a `cudf::size_type` (aka `int32_t`), you don't need to worry about copying as it is trivial for integer like types. When people type, `cudf::size_type size = std::distance(begin, end);` that is technically a narrowing conversion that is missed by the compiler because braced initialization is not being used. The `cudf::distance` avoids this. See: https://www.godbolt.org/z/s3ddKo
```cpp
    // this compiles fine, even though it 
    // should be a narrowing conversion
    cudf::size_type const sz = std::distance(f, l); 

    // this fails to compile, because braced 
    // initialization ensures narrowing conversions
    // are caught
    cudf::size_type const sz2{std::distance(f, l)}; 

    // this compiles fine, remove the "noise" of the 
    // static cast and ensures no narrowing conversion
    auto const sz3 = cudf::distance(f, l);
```

Reply:
You are talking about a different line. I was referring to the change from `std::vector<bitmask_type> null_mask(num_words, 0);` to `auto null_mask = std::vector<bitmask_type>(num_words, 0);`

Reply:
Oh i understand. This is a style preference because it is much more readable in my opinion, no perf difference. For example:
```cpp
// ctor calls without equal sign
std::vector<int> const vec(begin, begin + size);
cudf::test::fixed_width_column_wrapper const col{1, 2, 3, 4, 5};
cudf::size_type size{10};

// using ctor + equal sign = makes variable names much easier to parse
auto const vec  = std::vector<int>(begin, begin + size);
auto const col  = cudf::test::fixed_width_column_wrapper{1, 2, 3, 4, 5};
auto const size = cudf::size_type{10};
```
I find the variable names an order (or two) of magnitude easier to parse.